src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/Contract.cs

```diff
@@ -22,7 +22,20 @@ internal static class Contract
         /// all builds
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void ThrowIfNull<T>([NotNull] T value, [CallerLineNumber] int lineNumber = 0)
+        public static void ThrowIfNull<T>([NotNull] T value, [CallerLineNumber] int lineNumber = 0) where T : class?
```

Comment:
Could be a follow up, but `CallerArgumentExpression` is available now too, right?

Reply:
Actually I'll let this be a follow-up since I'm not sure what the current state of that is for our toolchain.